Src/GUI/Widgets/OutputFilesWidget.py

```diff
@@ -56,17 +79,31 @@ def get_widget_value(self):
                                     
         return (filename, formats)
     
-    def set_data(self, datakey):
+    def msg_input_file_set(self, message):
+
+        configurator, filename = message
+
+        if self._configurator.root is None:
+            return
 
-        basename = "output"
+        if self._configurator.root == configurator.name:
+            filenameWithoutExt = os.path.splitext(filename)[0]
+            outputFileWithoutExt = '%s_converted' % filenameWithoutExt

```

Comment:
I don't like the '_converted' part of the name; I always call my converted trajectories by the same name as the initial ones. Since during trajectory conversion the file extension changes, I think we could drop that part and just use the original name. What do you think?

Reply:
Yeap let's drop the _converted part

Reply:
wait a minute. If you try on your side to remove the _converted you will see that it triggers some nasty side effect of the get_unique_filename method. Not a bug because it is an expected behavior yet not desirable. Indeed, as the get_unique_filename method works only with basename without extensions, if you pick for instance the DL_POLY input file FIELD_methanol, you will end up with a proposed output file name of FIELD_methanol(1). The use of _converted suffix or whatever suffix would save us from this.

Reply:
Well, in that I case I would prefer to check using the extensions as well rather than keeping the _converted in. Also, now that you have mentioned DL_POLY, it might be a good idea to remove the FIELD_ part from the suggested file name since that's the equivalent to an extension.

Reply:
The problem with the extensions is that the selected ones are not known at the time the dialog is popped up. That's the exact reason why at this level I work only with the basename.

For The FIELD, I can not remove it because that would make the OutputFilesWidget not generic anymore

Reply:
The smallest intervention to the code to achieve the former is probably to generate the basenames like so:
```
basenames = [os.path.splitext(f)[0] for f in filesInDirectory
                 if os.path.splitext(f)[1] == '.nc' or os.path.splitext(f)[1] == 'tar']
```

Reply:
Moreover coming back to the extensions, if you select more than one extension, it can happen that one of the filename with the extension exists while the others not. What should we do in such case ? We are stuck

Reply:
Aah, I see what you mean. My solution is to use both, and select the higher number.

Reply:
In the modified code that you showed, I dislike the fact that the extension are hardcoded. We are losing the generality of the widget.

Reply:
If by unselecting you mean doing this,
![image](https://user-images.githubusercontent.com/78977041/162734095-82d7c5c3-9a6a-490d-8fd3-b901ed3c85e9.png)
then I got an error.

About the generality of our code; is it more important than user experience? I don't know much about this aspect of software development, but my instinct is to prioritise users over the code.

Reply:
I think we don't need to hardcode the formats; there is the `self._configurator.formats` that we can access.

Reply:
I was trying to get the formats through REGISTRY
```
temp = []
        for i in REGISTRY['formats']:
            temp.extend(i.extensions)
```
but it says that `REGISTRY['formats']` is an empty dictionary; do you know what's going on?

Reply:
I can not reproduce the error you describe on ubuntu focal.

Reply:
For the generality of the code, let's derive a brand new class dedicated to the converter output. The class can be standalone or a child class of the OutputFileWidget. That should do the job without having too specific code in general class. However, it still does not solve the problem of the formats.

Reply:
It's strange that you are not getting an error when you have no formats selected, because, looking at the code, there definitely should be an error. There is no line in `OutputFilesConfigurator` that sets a certain format when none were selected. Instead, there is a line (71) that raises an error. I have confirmed on Windows 10 and Ubuntu 21.

Reply:
Also, I got the issue with REGISTRY, I had a typo. It should be like this:
```
temp = []
for i in REGISTRY['format'].values():
    temp.extend(i.extensions)
```
This allows us to access all the output formats. The only issue is that the ASCII format creates a .tar file.

Reply:
Thanks for the hint. However, if we consider that the output file of a converter is a very special case as it is an output file with a single extension (i.e. nc), deriving a specific widget would make sense. For instance, it would not contain the format combobox as the output extension is by construction nc. What do you think ?

Reply:
yeap exactly